### ISSUE TITLE
Fails to compile with OpenSSL 1.1.0 or newer

### ISSUE THREAD

**Issue by nyuszika7h** [allinurl/gwsocket#28]
OpenSSL 1.1.0 introduces breaking changes which prevents gwsocket from compiling. This is what I did to get it to compile with OpenSSL 1.1.1 on Debian 10:

* `SSL_library_init` needs to be replaced with `SSL_SESSION_new`.
* `CRYPTO_num_locks` is now a macro, so the `AC_CHECK_LIB` needs to be removed.
* `-lcrypto` needs to be added to LDFLAGS. I did `./configure --with-openssl LDFLAGS=-lcrypto`.

**Comment by allinurl** [comment]
Thanks for reporting this! Would you be able to submit a PR?

**Comment by FlamingPaw** [comment]
I came across this issue as well, and applied the changes @nyuszika7h mentioned above on my fork at https://github.com/FlamingPaw/gwsocket.
I am still doing some testing on it, but it appears to be working properly with OpenSSL 1.1.0+

I also had to change the make command in addition to the compile command. The full install procedure is as follows:
$ `git clone https://github.com/FlamingPaw/gwsocket.git`
$ `cd gwsocket`
$ `autoreconf -fiv`
$ `./configure --with-openssl LDFLAGS=-lcrypto` <-- Updated
$ `make LIBS="-lssl -lcrypto"` <-- Updated
$ `make install`

Testing with `gwsocket --ssl-cert /path/to/my/ssl/cert --ssl-key /path/to/my/ssl/key --port 7891 --echo-mode` provides a successful wss connection with gwsocket properly responding by echoing the received messages.
